test_scripts/API/VehicleData/common.lua

```diff
@@ -455,17 +473,83 @@ function m.unexpectedDisconnect(pParam)
   utils.wait(1000)
 end
 
+--[[ @unexpectedDisconnect: Unexpected disconnect sequence for two apps for different <vd_param> parameters
+--! @parameters:
+--! pParam1: name of the VD parameter for 1st app
+--! pParam2: name of the VD parameter for 2st app
+--! @return: none
+--]]
+function m.unexpectedDisconnectTwoApps(pParam1, pParam2)
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+  :Times(2)
+  m.getHMIConnection():ExpectRequest("VehicleInfo.UnsubscribeVehicleData")
+  :ValidIf(function(_, data)
+    if data.params[pParam1] == true or data.params[pParam2] == true then
+      return true
+    else
+      return false, "VehicleInfo.UnsubscribeVehicleData request contains unexpected parameter" ..
+        utils.tableToString(data.params)
+    end
+  end)
+  :Times(2)
+  actions.mobile.disconnect()
+  utils.wait(1000)
+end
+
+--[[ @unregisterApp: Unregister App sequence
+--! @parameters:
+--! pAppId: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.unregisterAppWithUnsubscription(pParam, pAppId, isRequestOnHMIExpected)
+  if pAppId == nil then pAppId = 1 end
+  if isRequestOnHMIExpected == nil then isRequestOnHMIExpected = true end
+  local response = {
+    dataType = m.vd[pParam],
+    resultCode = "SUCCESS"
+  }
+  local responseParam = pParam
+  if pParam == "clusterModeStatus" then responseParam = "clusterModes" end
+  if isRequestOnHMIExpected == true then
+    m.getHMIConnection():ExpectRequest("VehicleInfo.UnsubscribeVehicleData", { [pParam] = true })
+    :Do(function(_,data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { [responseParam] = response })
+    end)
+  else
+    m.getHMIConnection():ExpectRequest("VehicleInfo.UnsubscribeVehicleData"):Times(0)
+  end
+  local mobileSession = m.getMobileSession(pAppId)
+  local cid = mobileSession:SendRPC("UnregisterAppInterface",{})
+  actions.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered",
+    { unexpectedDisconnect = false })
+  mobileSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS"})
+end
```

Comment:
Can we use existing `processSubscriptionRPC()` inside of this function in order not to have code duplicates?

Reply:
I suppose not due to redundant UnsubscribeVehicleData RPC in existing processSubscriptionRPC () 
local cid = m.getMobileSession(pAppId):SendRPC(pRPC, { [pParam] = true })